File: leetcode_solutions/0055-jump-game/solution.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        max_reach = 0
        n = len(nums)
        for i in range(n):
            if i > max_reach:
                return False
            max_reach = max(max_reach, nums[i] + i)
            
        return max_reach >= n - 1

File: leetcode_solutions/0055-jump-game/solution.java
class Solution {
    public boolean canJump(int[] nums) {
        int maxReach = 0;
        for (int i = 0; i < nums.length - 1; i++){
            if (i > maxReach){
                return false;
            }
            maxReach = Math.max(nums[i] + i, maxReach);
        }
        return maxReach >= nums.length - 1;
    }
}
